File: app/src/main/java/com/mxt/anitrend/base/custom/view/widget/CustomRatingBar.java
package com.mxt.anitrend.base.custom.view.widget;

import android.content.Context;
import androidx.databinding.BindingAdapter;
import androidx.appcompat.widget.AppCompatRatingBar;
import android.util.AttributeSet;

import com.mxt.anitrend.base.interfaces.view.CustomView;

/**
 * Created by max on 2017/10/27.
 * Custom rating bar
 */

public class CustomRatingBar extends AppCompatRatingBar implements CustomView {

    public CustomRatingBar(Context context) {
        super(context);
        onInit();
    }

    public CustomRatingBar(Context context, AttributeSet attrs) {
        super(context, attrs);
        onInit();
    }

    public CustomRatingBar(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        onInit();
    }

    /**
     * Optionally included when constructing custom views
     */
    @Override
    public void onInit() {
        setIsIndicator(true);
        setStepSize(0.1f);
        setNumStars(5);
        setMax(5);
    }

    /**
     * Clean up any resources that won't be needed
     */
    @Override
    public void onViewRecycled() {

    }

    @BindingAdapter("averageScore")
    public static void setAverageScore(CustomRatingBar view, int meanScore) {
        float rating = (float)meanScore * view.getMax() / 100;
        view.setRating(rating);
    }
}

File: app/src/main/java/com/mxt/anitrend/base/custom/view/widget/CustomRatingBar.kt
package com.mxt.anitrend.base.custom.view.widget

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatRatingBar
import com.mxt.anitrend.base.interfaces.view.CustomView

/**
 * Created by max on 2017/10/27.
 * Custom rating bar
 */
class CustomRatingBar @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatRatingBar(context, attrs, defStyleAttr), CustomView {

    init {
        onInit()
    }

    /**
     * Optionally included when constructing custom views
     */
    override fun onInit() {
        setIsIndicator(true)
        stepSize = 0.1f
        numStars = 5
        max = 5
    }

    /**
     * Clean up any resources that won't be needed
     */
    override fun onViewRecycled() = Unit

    companion object {
        @JvmStatic
        fun setAverageScore(view: CustomRatingBar, meanScore: Int) {
            val rating = meanScore * view.max / 100f
            view.rating = rating
        }
    }
}
